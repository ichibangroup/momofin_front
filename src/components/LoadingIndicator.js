import React from 'react';

const LoadingIndicator = () => (
  <div className="flex justify-center items-center h-full" data-testid="loading-indicator">
    <div
      className="animate-spin rounded-full h-12 w-12 border-t-2 border-b-2 border-blue-500"
      data-testid="spinner"
      role="status"
      aria-label="Loading"
    />
  </div>
);

export default LoadingIndicator;