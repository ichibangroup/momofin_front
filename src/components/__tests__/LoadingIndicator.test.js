import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import LoadingIndicator from '../LoadingIndicator';

describe('LoadingIndicator', () => {
  it('renders without crashing', () => {
    render(<LoadingIndicator />);
    expect(screen.getByTestId('loading-indicator')).toBeInTheDocument();
  });

  it('has correct structure and classes', () => {
    render(<LoadingIndicator />);
    const outerDiv = screen.getByTestId('loading-indicator');
    const innerDiv = screen.getByTestId('spinner');

    expect(outerDiv).toHaveClass('flex', 'justify-center', 'items-center', 'h-full');
    expect(innerDiv).toHaveClass('animate-spin', 'rounded-full', 'h-12', 'w-12', 'border-t-2', 'border-b-2', 'border-blue-500');
  });

  it('is accessible', () => {
    render(<LoadingIndicator />);
    const loadingIndicator = screen.getByRole('status');
    expect(loadingIndicator).toHaveAccessibleName('Loading');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<LoadingIndicator />);
    expect(asFragment()).toMatchSnapshot();
  });
});