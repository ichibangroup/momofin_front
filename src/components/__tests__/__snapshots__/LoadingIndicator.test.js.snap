// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center h-full"
  >
    <div
      class="animate-spin rounded-full h-12 w-12 border-t-2 border-b-2 border-blue-500"
    />
  </div>
</DocumentFragment>
`;
